native/src/render/nova_renderer.cpp

```diff
@@ -494,6 +500,15 @@ namespace nova {
 
     void nova_renderer::end_frame() {
         NOVA_PROFILER_FLUSH_TO_FILE("profiler_data.txt");
+        std::vector<uint32_t> *buffer = new std::vector<uint32_t>(passes_list.size() * 2);
+        context->device.getQueryPoolResults(context->timestamp_query_pool, 0, buffer->size(), buffer->size() *
+                                                    sizeof(uint32_t), buffer->data(), sizeof(uint32_t), vk::QueryResultFlagBits::eWait);
+        unsigned int index = 0;
+        while(index < (buffer->size() - 1)) {
+            uint32_t beginVal = buffer->at(index++);
+            uint32_t endVal = buffer->at(index++);
+            LOG(TRACE) << "Pass " << passes_list.at((index / 2) - 1).name << " took " << (endVal - beginVal);
```

Comment:
The value you get back from the GPU isn't exactly the number of nanoseconds the pass took. You have to multiply the difference here by `VkPhysicalDeviceProperties.limits.timestampPeriod`. We should be storing a `VkPhysicalDeviceProperties` for the current GPU in the `render_context`